### PR TITLE
support scheduled event images

### PR DESCRIPTION
**Pull request by akiraveliara**
this is based off #1197, with very minimal changes

this also calls for improvements to ImageTool.cs, since it's now actually used in corelib (this isn't a multipart request for some unfathomable reason, so we end up needing the image tool) - subject for a future PR

### CODE REVIEW

File: DSharpPlus.Rest/DiscordRestClient.cs [hunk 1]
```diff
@@ -54,8 +54,8 @@ public async Task InitializeCacheAsync()
     /// <param name="end">When the event ends. Required for <see cref="ScheduledGuildEventType.External"/></param>
     /// <param name="location">Where this location takes place.</param>
     /// <returns>The created event.</returns>
-    public async Task<DiscordScheduledGuildEvent> CreateScheduledGuildEventAsync(ulong guildId, string name, string description, ulong? channelId, ScheduledGuildEventType type, ScheduledGuildEventPrivacyLevel privacyLevel, DateTimeOffset start, DateTimeOffset? end, string location = null)
-        => await this.ApiClient.CreateScheduledGuildEventAsync(guildId, name, description, start, type, privacyLevel, new DiscordScheduledGuildEventMetadata(location), end, channelId);
+    public async Task<DiscordScheduledGuildEvent> CreateScheduledGuildEventAsync(ulong guildId, string name, string description, ulong? channelId, ScheduledGuildEventType type, ScheduledGuildEventPrivacyLevel privacyLevel, DateTimeOffset start, DateTimeOffset? end, Stream? image = null, string location = null)
```
Comment (InFTord) [inline DSharpPlus.Rest/DiscordRestClient.cs:57]: where's param docs for Stream :german:

Comment (akiraveliara) [reply DSharpPlus.Rest/DiscordRestClient.cs:57]: DiscordRestClient can die a thousand deaths

**Review by Plerx2493 — COMMENTED** [review]
LGTM

File: DSharpPlus.Rest/DiscordRestClient.cs (hunk 1 above)
Comment (InFTord) [reply DSharpPlus.Rest/DiscordRestClient.cs:57]: you will die a thousand and one deaths if you don't fix that /j

**Pull request merged**